### PR TITLE
Use $ldflags in ./configure macro in order to set LDFLAGS.

### PR DESCRIPTION
**Pull request by marxin**

### CODE REVIEW

File: macros.in [hunk 1]
```diff
@@ -1021,6 +1021,7 @@ package or when debugging this package.\
   CFLAGS="${CFLAGS:-%optflags}" ; export CFLAGS ; \
   CXXFLAGS="${CXXFLAGS:-%optflags}" ; export CXXFLAGS ; \
   FFLAGS="${FFLAGS:-%optflags}" ; export FFLAGS ; \
+  LDFLAGS="${LDFLAGS:-%?ldflags}" ; export LDFLAGS; \
```
Comment (Conan-Kudo) [inline macros.in]: But... we don't have `%ldflags`? I know this should initialize to an empty variable, but I don't know how that gets interpreted by autotools (or things pretending to offer autotools-like interfaces).

Comment (marxin) [reply macros.in]: Well, I would expect the `autotools` should be fine with empty `LDFLAGS`.

**Comment by marxin** [comment]
Motivation for the change:

I'm planning to enable LTO in `openSUSE:Factory` and I need to add `-flto=N` to LDFLAGS in order to make LTO linking phase parallel. That speeds up package build significantly.

**Comment by pmatilai** [comment]
No objection to adding LDFLAGS to %configure, it's actually long overdue. Fedora has had its own version of this for a quite some time now, but nobody got around to upstreaming it. This seems like as good time as any to take a look at that and consider which parts would be upstreamable.

There are a whole bunch of related things done, mostly done in redhat-rpm-config. For details, see https://src.fedoraproject.org/rpms/redhat-rpm-config/blob/master/f/macros#_27 but in summary:
- %optflags is only a building block for language specific %build_fooflags macros (except for ldflags which is not related)
- the %build_fooflags macros are used for setting the environment
- the environment setting is separated from %configure so it's usable for other make systems too
- there are further %extension_fooflags macros which are meant for building things like perl and python C-extensions

In addition, %build_ldflags is passed as RPM_LD_FLAGS to all the build scripts via %___build_pre, but that's an older addition than the redhat-rpm-config stuff above so not sure how relevant/useful it is in reality.

**Comment by marxin** [comment]
> No objection to adding LDFLAGS to %configure, it's actually long overdue. Fedora has had its own version of this for a quite some time now, but nobody got around to upstreaming it. This seems like as good time as any to take a look at that and consider which parts would be upstreamable.

Would you be fine with this pull request as is?

> 
> There are a whole bunch of related things done, mostly done in redhat-rpm-config. For details, see https://src.fedoraproject.org/rpms/redhat-rpm-config/blob/master/f/macros#_27 but in summary:
> 
> * %optflags is only a building block for language specific %build_fooflags macros (except for ldflags which is not related)
> * the %build_fooflags macros are used for setting the environment
> * the environment setting is separated from %configure so it's usable for other make systems too
> * there are further %extension_fooflags macros which are meant for building things like perl and python C-extensions
> 
> In addition, %build_ldflags is passed as RPM_LD_FLAGS to all the build scripts via %___build_pre, but that's an older addition than the redhat-rpm-config stuff above so not sure how relevant/useful it is in reality.

I'm CCing SUSE related people how are much more skilled with that: @scarabeusiv @DimStar77 @mlschroe

**Comment by pmatilai** [comment]
> Would you be fine with this pull request as is?

Sort of. I mean, I'm not expecting *you* to merge all that Fedora etc stuff in, but I'd rather discuss the direction and naming (eg %build_ldflags vs %ldflags) a bit before merging anything at all.

**Comment by ignatenkobrain** [comment]
I think switching to `%{build_…}` convention makes much more sense. In the old times where `optflags` meant everything, nowadays you want to have different CFLAGS, CXXFLAGS. And having `%{build_cflags}` but `%{ldflags}` sounds inconsistent.

**Comment by marxin** [comment]
> I think switching to `%{build_…}` convention makes much more sense. In the old times where `optflags` meant everything, nowadays you want to have different CFLAGS, CXXFLAGS. And having `%{build_cflags}` but `%{ldflags}` sounds inconsistent.

I'm fine with the `%build_ldflags` name which is aligned with the `redhat-rpm-config` name conventions.

**Comment by Conan-Kudo** [comment]
As one of the maintainers of [`rpm-config-SUSE`](https://github.com/openSUSE/rpm-config-SUSE), I'm okay with this idea and we could work out a plan for transitioning things to the convention in `redhat-rpm-config`. I like them more than the ones in the SUSE environment, as it's clearer what they are for.

**Comment by marxin** [comment]
OK, I've updated the pull request based on the discussion that we've got so far.

**Comment by marxin** [comment]
May I please ping this..

**Comment by pmatilai** [comment]
Just submitted a somewhat more elaborate update to the build flags (and yes including ldflags) system: #692

**Comment by marxin** [comment]
Ok, I'm closing the pull request.

**Pull request closed without merging**